File: 1. Easy/448. Find All Numbers Disappeared in an Array/Solution.java
class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        List<Integer> res = new ArrayList<Integer>();
        
        for (int i = 0; i < nums.length; i++) {
            int x = Math.abs(nums[i]) - 1;
            nums[x] = -Math.abs(nums[x]); 
        }

        for (int j = 0; j < nums.length; j++)
            if(nums[j] > 0)
                res.add(j + 1);

        return res;
    }
}

File: 1. Easy/448. Find All Numbers Disappeared in an Array/solution.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var findDisappearedNumbers = function (nums) {
  for (let i = 0; i < nums.length; i++) {
    let j = nums[i] - 1;
    while (i < nums.length && nums[i] != nums[j]) {
      j = nums[i] - 1;
      [nums[i], nums[j]] = [nums[j], nums[i]];
    }
  }

  let res = [];
  for (let i = 0; i < nums.length; i++)
    if (nums[i] != i + 1)
        res.push(i + 1);

  return res;
};
